### PR TITLE
feat: Add a way to set labels

### PR DESCRIPTION
**Pull request by vv9k**
This PR reworks the existing `EnvVarRow` to a more generic `KeyValRow` in order to be reused for labels. For now it has a predefined header labels `Key` and `Value` as I think it fits both env vars and labels so if there is a need to change the headers it might be implemented in the future PRs.

image

Closes: #334

### CODE REVIEW

File: data/resources/resources.gresource.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
     <file compressed="true" preprocess="xml-stripblanks">ui/containers/count-bar.ui</file>
     <file compressed="true" preprocess="xml-stripblanks">ui/containers/group.ui</file>
     <file compressed="true" preprocess="xml-stripblanks">ui/containers/panel.ui</file>
-    <file compressed="true" preprocess="xml-stripblanks">ui/env-var/row.ui</file>
+    <file compressed="true" preprocess="xml-stripblanks">ui/key-val/row.ui</file>
```
Comment (marhkb) [inline data/resources/resources.gresource.xml:33]: This line needs to be moved a few lines down to keep the list sorted.

**Review by marhkb — CHANGES_REQUESTED** [review]

**Comment by marhkb** [comment]
Thank you. :-)

Almost LGTM. 
I've just one small change request regarding the code and one regarding the commit messages. Could you tag the commits like for example:
```
refactor(env-var-row): Transform it into a more generic KeyValRow
feat(container-creation-page): Add a widget for defining labels
```

It doesn't matter if the length exceeds 50 characters, up to [72](https://wiki.gnome.org/Git/CommitMessages) are ok for me.

**Comment by vv9k** [comment]
Done, made the list sorted and fixed the commit messages.

**Comment by vv9k** [comment]
I was also considering expanding the image build page as well as pods creation page to also include the labels. Do you think it would be best to also introduce a AdwViewStack in image build page and put labels in the integration tab to match the container creation page?

File: data/resources/ui/image/build-page.ui [hunk 2]
```diff
@@ -54,52 +53,132 @@
...
+
+                                  </object>
+                                </property>
+
+                              </object>
+                            </child>
+
                           </object>
                         </child>
-
```
Comment (marhkb) [inline data/resources/ui/image/build-page.ui:96]: The blank space should be kept here.

File: data/resources/ui/image/build-page.ui [hunk 3]
```diff
@@ -54,52 +53,132 @@
...
                       </object>
                     </child>
 
                   </object>
                 </child>
-
```
Comment (marhkb) [inline data/resources/ui/image/build-page.ui:102]: The blank space should be kept here.

**Review by marhkb — CHANGES_REQUESTED** [review]

**Comment by marhkb** [comment]
> I was also considering expanding the image build page as well as pods creation page to include the labels as well. Do you think it would be best to also introduce a AdwViewStack in image build page and put labels in the integration tab to match the container creation page?

LGTM. Nice work. This can land if you resolve the two small change requests I made and the conflict.

**Comment by vv9k** [comment]
Awesome :) fixed the outstanding issues with whitespace so should be ready for merge.

**Review by marhkb — APPROVED** [review]

**Comment by marhkb** [comment]
Thanks :-)

**Pull request merged**